lib/util.js

```diff
@@ -993,6 +997,29 @@ function formatPromise(ctx, value, recurseTimes, keys) {
   return output;
 }
 
+function formatNamespaceProperty(ctx, ns, recurseTimes, key, array) {
+  let name;
+  let value;
+  try {
+    value = formatValue(ctx, ns[key], recurseTimes, array === 0);
+  } catch (err) {
+    if (err instanceof ReferenceError) {
+      value = ctx.stylize('<uninitialized>', 'special');
```

Comment:
I am somewhat surprised that the keys are exposed even though accessing them throws. I guess that is done by design - out of curiosity: would you be so kind and point me to the reasons why this is done like that?

Reply:
even though we know the export names, they aren't yet intialised (think `console.log(a); const a = 1`). there is no js value that can represent this state so v8 throws a reference error, which is what would happen with that example.